vpr/src/route/connection_router.cpp

```diff
@@ -428,17 +484,19 @@ void ConnectionRouter<Heap>::timing_driven_expand_neighbour(t_heap* current,
                                                             const t_conn_cost_params cost_params,
                                                             const t_bb bounding_box,
                                                             int target_node,
-                                                            const t_bb target_bb) {
+                                                            const t_bb target_bb,
+                                                            std::set<int>* route_tree_nodes) {
     RRNodeId to_node(to_node_int);
     int to_xlow = rr_nodes_.node_xlow(to_node);
     int to_ylow = rr_nodes_.node_ylow(to_node);
     int to_xhigh = rr_nodes_.node_xhigh(to_node);
     int to_yhigh = rr_nodes_.node_yhigh(to_node);
 
-    if (to_xhigh < bounding_box.xmin      //Strictly left of BB left-edge
-        || to_xlow > bounding_box.xmax    //Strictly right of BB right-edge
-        || to_yhigh < bounding_box.ymin   //Strictly below BB bottom-edge
-        || to_ylow > bounding_box.ymax) { //Strictly above BB top-edge
+    if ((to_xhigh < bounding_box.xmin    //Strictly left of BB left-edge
+         || to_xlow > bounding_box.xmax  //Strictly right of BB right-edge
+         || to_yhigh < bounding_box.ymin //Strictly below BB bottom-edge
+         || to_ylow > bounding_box.ymax)
+        && (route_tree_nodes == nullptr)) { //Strictly above BB top-edge
```

Comment:
Why did you remove BB pruning if `route_tree_nodes` is present?

Also the comment is no longer aligned with the relevant logic

Reply:
I guess this is another hard to read one, it's to check if RCV is enabled. It should be using the run_rcv bool I created below it. This just makes it easier for RCV to resolve particularly difficult nets, but looking at this again now there is opportunity to be smarter and only do this for nets that might have issues. Such a check could speed up routing with RCV enabled.

Reply:
Your reply doesn't explain why RCV disables the BB pruning?  I can understand RCV doing a BB expansion, but disabling the BB seems unwise?